litestar/events/emitter.py

```diff
@@ -77,19 +79,25 @@ def __init__(self, listeners: Sequence[EventListener]) -> None:
         self._send_stream: MemoryObjectSendStream | None = None
         self._exit_stack: AsyncExitStack | None = None
 
-    @staticmethod
-    async def _worker(receive_stream: MemoryObjectReceiveStream) -> None:
+    async def _worker(self, receive_stream: MemoryObjectReceiveStream) -> None:
         """Run items from ``receive_stream`` in a task group.
 
         Returns:
             None
         """
-        async with receive_stream, anyio.create_task_group() as task_group:
+        async with receive_stream:
             async for item in receive_stream:
-                fn, args, kwargs = item
+                await self._run_listener_in_task_group(*item)
+
+    @staticmethod
+    async def _run_listener_in_task_group(fn: Any, args: tuple[Any], kwargs: dict[str, Any]) -> None:
```

Comment:
This solution is a bit of an issue unfortunately, as creating a new task group for every received item will prevent them running concurrently. I think the proper fix here would be to wrap the listeners in an exception handler and then run those in the task group

Reply:
@peterschutt I can cook up a PR for this later unless you want to fix it :)

Reply:
OK, I'll leave it to you - thanks.

Reply:
Actually, I got it - my mess, I'll clean it up.